server/api.js

```diff
@@ -364,33 +375,103 @@ router.get("/tenders", async (req, res) => {
 	const offset = (page - 1) * limit;
 
 	const countSql = "SELECT COUNT(*) FROM tender";
-	const dataSql = `
-		SELECT id, title, creation_date, announcement_date, deadline, description, status
+	const tendersSql = `
+		SELECT id, title, creation_date, announcement_date, deadline, description, status, closing_date
 		FROM tender 
 		ORDER BY creation_date DESC 
 		LIMIT $1 OFFSET $2
 	`;
+
 	try {
 		const countResult = await db.query(countSql);
 		const totalItems = parseInt(countResult.rows[0].count, 10);
 		const totalPages = Math.ceil(totalItems / limit);
 
-		const dataResult = await db.query(dataSql, [limit, offset]);
-		const tenders = dataResult.rows;
+		const tendersResult = await db.query(tendersSql, [limit, offset]);
+		const tenders = tendersResult.rows;
 
-		res.status(200).json({
-			results: tenders,
-			pagination: {
-				itemsPerPage: limit,
-				currentPage: page,
-				totalPages,
-			},
-		});
+		const tenderIds = tenders.map((tender) => tender.id);
+
+		if (tenderIds.length > 0) {
+			const bidsSql = `
+				SELECT *
+				FROM bid
+				WHERE tender_id = ANY($1::int[])
```

Comment:
nit. You didn't need ANY here. IN was enough.

```suggestion
				WHERE tender_id IN($1::int[])
```